NewKernel_d/include/CGAL/NewKernel_d/function_objects_cartesian.h

```diff
@@ -1170,6 +1193,41 @@ template<class R_> struct Compare_lexicographically : private Store_kernel<R_> {
 
 CGAL_KD_DEFAULT_FUNCTOR(Compare_lexicographically_tag,(CartesianDKernelFunctors::Compare_lexicographically<K>),(),(Construct_ttag<Point_cartesian_const_iterator_tag>));
 
+namespace CartesianDKernelFunctors {
+template<class R_> struct Compare_squared_distance : private Store_kernel<R_> {
+        CGAL_FUNCTOR_INIT_STORE(Compare_squared_distance)
+        typedef R_ R;
+        typedef typename Get_type<R, RT_tag>::type RT;
+        typedef typename Get_type<R, Comparison_result_tag>::type result_type;
+        typedef typename Get_functor<R, Construct_ttag<Point_cartesian_const_iterator_tag> >::type CI;
+        // TODO: This is_exact thing should be reengineered.
+        // the goal is to have a way to tell: don't filter this
+        typedef typename CGAL::Uses_no_arithmetic<CI> Uses_no_arithmetic;
+
+        template<class V,class W>
+        result_type operator()(V const&a, V const&b, W const&c)const{   //  Point, Point. FT
+                CI ci(this->kernel());
+
+                auto a_begin=ci(a,Begin_tag());
+                auto b_begin=ci(b,Begin_tag());
+                auto a_end=ci(a,End_tag());
+                RT sqdist(0);
+                result_type res;
+                // can't we do slightly better for Uncertain<*> ?
+                // after res=...; if(is_uncertain(res))return indeterminate<result_type>();
+                do{
+                  RT d = (*a_begin++ - *b_begin++);
+                  sqdist += d*d;
+                }while(a_begin!=a_end);
```

Comment:
If you decided to compare sqdist to c after each iteration of the loop (just a random example, not a suggestion), it would make sense to have special code. But if you just compute the squared distance, why not to use `typedef typename Get_functor<R, Squared_distance_tag>::type CSD;` like `Compare_distance` does?

Reply:
changed.